client/signup/config/flows.js

```diff
@@ -128,7 +128,7 @@ const flows = {
 
 	developer: {
 		steps: [ 'themes', 'site', 'user' ],
-		destination: '/me/next?welcome',
+		destination: '/devdocs/welcome',
```

Comment:
I checked with @scruffian and a signup flow only shows up when you link directly to the flow (i.e., via `/start/developer`) and that only gets linked to from `devdocs/controller.js`... which is only available in an environment with `devdocs` enabled. So I don't think any other checks are necessary at this point.